### DISCUSSION TITLE
Removing @ from the url

### DISCUSSION THREAD

**Discussion by ugintl** [JulianPrieber/littlelink-custom#152] [General]
I want to remove @ from the url. I have edited few files and it is working perfectly. Please see the file names below. Is it all or do I need to make changes on other places as well?

1. routes/web.php
2. resources/views/panel/edit-user.blade.php
3. resources/views/layouts/sidebar.blade.php
4. resources/views/studio/page.blade.php
5. resources/views/panel/users.blade.php

**Comment by JulianPrieber** [comment]
Um... :grimacing: 

There is an option for that in the advanced config.

**Comment by ugintl** [comment]
@JulianPrieber oh I saw the advanced config. but I did not understand. Can you give example how I would do that? I want to remove @ from user profile and from the "page" section as well where the user creates the profile page..

**Reply by JulianPrieber** [reply]
It's this option in your advanced config:
`'custom_url_prefix' => '+', // The '@' prefix will always work regardless of this setting.`

Mainly, this is designed to allow to use a custom prefix like the '+' in this case. If you empty this filed, you can access littlelink pages without the prefix.

This is a bit buggy tho and will probably break your @ prefix. This will not replace the prefix on any other page, it's made to simply be an alternative to the @ not replace it.

If you change everything you set out to change, removing the @ from the source code, you might be able to use this natively this way.


However, just as a heads-up if a user names themselves something like a route (for example 'update') you won't be able to access that page.
This is the reason why the prefix is there in the first place.

**Reply by ugintl** [reply]
WHat I have understood from your message is that my changes to the above mentioned files is the best workaround right now? As you have said, it is buggy.

As for username clash with route, We can disallow some usernames which might clash with the routes. There wont be too many I guess. What if I completely want to remove @? Is what I did above enough? Dont forget there are metatags as well.

 More importantly, when I update the system to new version, all changes will be lost? Not sure of that.

**Reply by JulianPrieber** [reply]
Yeah, so if you're the only one using your site, and you just want to use one of your pages on your socials, you can easily use only the advanced config.

Otherwise, you will probably have to go through everything you mentioned first.
I never really tried anything like this, so you'll have to try what works.

I'm planning on releasing an update sometime that can do this natively.

And yes, if you go ahead and modify the source code, these changes will be undone after you update.

**Reply by ugintl** [reply]
ok thanks

**Reply by ugintl** [reply]
one more thing. Generally, When we want to make a change in the design, instead of editing original css files, we make a custom css file. So is there something like this in laravel that instead of customizing the original blade files, you make a custom blade file and every change goes in it. Would it not be more clean ? Just a thought

**Reply by JulianPrieber** [reply]
I would love something like a plugin system that can do something like this as an extension.

However, I don't know of any system that can easily implement this.
Moreover, frequent changes in these files would make such a system difficult to implement and even if managed difficult to maintain. 

I'm really open to such a concept, but I don't think something like this will be implemented any time soon.
